packages/react-ui-core/src/Gmap/Gmap.js

```diff
@@ -1,67 +1,49 @@
 import React, { PureComponent } from 'react'
 import PropTypes from 'prop-types'
-import themed from 'react-themed'
-import { parseArgs } from '@rentpath/react-ui-utils'
+import { GoogleApiWrapper } from 'google-maps-react'
 import GmapBase from './GmapBase'
 
-const createComponent = (element, className) => {
-  if (React.isValidElement(element)) return element
-
-  const [Container, props] = parseArgs(element, 'div', { className })
-  return <Container {...props} />
-}
-
-@themed(/^Gmap/, { pure: true })
-
 export default class Gmap extends PureComponent {
   static propTypes = {
-    loadingElement: PropTypes.oneOfType([
-      PropTypes.object,
-      PropTypes.element,
-      PropTypes.node,
-    ]),
-    containerElement: PropTypes.oneOfType([
-      PropTypes.object,
-      PropTypes.element,
-    ]),
-    mapElement: PropTypes.oneOfType([
-      PropTypes.object,
-      PropTypes.element,
-    ]),
-    theme: PropTypes.object,
+    apiKey: PropTypes.string.isRequired,
+    language: PropTypes.string,
+    libraries: PropTypes.array,
+    version: PropTypes.string,
   }
 
   static defaultProps = {
-    theme: {},
+    version: '3',
   }
 
-  get containerElement() {
-    const { containerElement, theme } = this.props
-    return createComponent(containerElement, theme.Gmap_Container)
+  get api() {
+    const {
+      apiKey,
+      language,
+      libraries,
+      version,
+    } = this.props
+
+    return GoogleApiWrapper({
+      apiKey,
+      language,
+      libraries,
+      version,
+    })(GmapBase)
   }
 
-  get mapElement() {
-    const { mapElement, theme } = this.props
-    return createComponent(mapElement, theme.Gmap_Element)
-  }
+  render() {
+    const {
+      apiKey,
```

Comment:
do these 4 props need to be here?

Reply:
Yeah because I combined all props into one component so I'd rather not pass those on.